rules/framework/framework_packaging.py

```diff
@@ -80,6 +80,40 @@ def _copy_modulemap(framework_root, modulemap_path):
     _mkdir(os.path.dirname(dest))
     _cp(modulemap_path, dest)
 
+def _clean(framework_root, manifest_file, output_manifest_file):
+    """Remove stale files from the framework root.
+
+    Args:
+        framework_root: root folder of the framework
+        manifest_file: path to the input manifest file
+        output_manifest_file: path to the output manifest file
+    """
+
+    with open(manifest_file) as f:
```

Comment:
If this runs in full builds as well as incremental, I'd implement a _fail early_ here where if the `framework_root` doesn't exist this function just returns without having to read the manifest file.

Reply:
How about adding? 
```suggestion
    if not os.path.exists(framework_root):
        return
    with open(manifest_file) as f:
```?